### PR TITLE
[OADP-1631] Velero CSI plugin now determines whether to restore Volume’s data from snapshots on the restore’s restorePVs setting.

### PR DESCRIPTION
**Pull request by CarmiWisemon**
OADP 1.2.0

OCP - 4.10 +

Resolves - https://issues.redhat.com/browse/OADP-1631

Deploy preview - https://59780--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.html

### CODE REVIEW

File: modules/oadp-creating-restore-cr.adoc [hunk 1]
```diff
@@ -36,10 +36,11 @@ spec:
   - backups.velero.io
   - restores.velero.io
   - resticrepositories.velero.io
-  restorePVs: true
+  restorePVs: true <3>
 ----
 <1> Name of the `Backup` CR.
 <2> Optional. Specify an array of resources to include in the restore process. Resources might be shortcuts (for example, 'po' for 'pods') or fully-qualified. If unspecified, all resources are included.
+<3> restorePVs can be set to `false` in order to turn off backup of `PersistentVolumes`.
```
Comment (PrasadJoshi12) [inline modules/oadp-creating-restore-cr.adoc:43]: It's an optional field.

**Comment by kaovilai** [comment]
/retitle [OADP-1631] Velero CSI plugin now determines whether to restore Volume’s data from snapshots on the restore’s restorePVs setting.

File: modules/oadp-creating-restore-cr.adoc (hunk 1 above)
Comment (kaovilai) [inline modules/oadp-creating-restore-cr.adoc:43]:
```suggestion
<3> restorePVs can be set to `false` in order to turn off restore of `PersistentVolumes`.
```
/cc @shubham-pampattiwar

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: modules/oadp-creating-restore-cr.adoc (hunk 1 above)
Comment (shubham-pampattiwar) [reply modules/oadp-creating-restore-cr.adoc:43]: Nope, it only specifies whether to restore the PVs from snapshots

Comment (CarmiWisemon) [reply modules/oadp-creating-restore-cr.adoc:43]: @shubham-pampattiwar @kaovilai 
Please explain what I should write for the description.

**Comment by CarmiWisemon** [comment]
@shubham-pampattiwar @kaovilai @PrasadJoshi12 

I updated the PR based on your comments, please review.

File: modules/oadp-creating-restore-cr.adoc (hunk 1 above)
Comment (shubham-pampattiwar) [reply modules/oadp-creating-restore-cr.adoc]: Maybe re-frame to:  `restorePVs` can be set to `false` in order to turn off restore of `PersistentVolumes` from `VolumeSnapshots`.

File: modules/oadp-creating-restore-cr.adoc [hunk 2]
```diff
@@ -36,10 +36,11 @@ spec:
   - backups.velero.io
   - restores.velero.io
   - resticrepositories.velero.io
-  restorePVs: true
+  restorePVs: true <3>
 ----
 <1> Name of the `Backup` CR.
 <2> Optional. Specify an array of resources to include in the restore process. Resources might be shortcuts (for example, 'po' for 'pods') or fully-qualified. If unspecified, all resources are included.
+<3> Optional: `restorePVs` can be set to `false` in order to turn off restore of `PersistentVolumes` from a `VolumeSnapshot`.
```
Comment (kaovilai) [inline modules/oadp-creating-restore-cr.adoc:43]:
```suggestion
<3> Optional: `restorePVs` can be set to `false` in order to turn off restore of `PersistentVolumes` from a `VolumeSnapshot` for CSI, or from native snapshots when VolumeSnapshotLocation is setup.
```

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: modules/oadp-creating-restore-cr.adoc [hunk 3]
```diff
@@ -36,10 +36,11 @@ spec:
   - backups.velero.io
   - restores.velero.io
   - resticrepositories.velero.io
-  restorePVs: true
+  restorePVs: true <3>
 ----
 <1> Name of the `Backup` CR.
 <2> Optional. Specify an array of resources to include in the restore process. Resources might be shortcuts (for example, 'po' for 'pods') or fully-qualified. If unspecified, all resources are included.
+<3> Optional: `restorePVs` can be set to `false` in order to turn off restore of `PersistentVolumes` from `VolumeSnapshot` of Container Storage Interface (CSI) snapshots, or from native snapshots when `VolumeSnaphsotLocation` is configured.
```
Comment (skopacz1) [inline modules/oadp-creating-restore-cr.adoc:43]: Typo: `VolumeSnaphsotLocation` -> `VolumeSnapshotLocation`

Comment (skopacz1) [inline modules/oadp-creating-restore-cr.adoc:43]: Suggestion:

```suggestion
<3> Optional: The `restorePVs` field can be set to `false` in order to turn off restore of `PersistentVolumes` from `VolumeSnapshot` of Container Storage Interface (CSI) snapshots, or from native snapshots when `VolumeSnaphsotLocation` is configured.
```

(If you implement this change, please avoid clicking "accept suggestion" and instead make this change locally)

Comment (skopacz1) [inline modules/oadp-creating-restore-cr.adoc:43]: Suggestion: you can probably drop the comma after "snapshots"

**Review by skopacz1 — COMMENTED** [review]
A few suggestions, otherwise looks good to me!

File: modules/oadp-creating-restore-cr.adoc [hunk 4]
```diff
@@ -36,10 +36,11 @@ spec:
   - backups.velero.io
   - restores.velero.io
   - resticrepositories.velero.io
-  restorePVs: true
+  restorePVs: true <3>
 ----
 <1> Name of the `Backup` CR.
-<2> Optional. Specify an array of resources to include in the restore process. Resources might be shortcuts (for example, 'po' for 'pods') or fully-qualified. If unspecified, all resources are included.
+<2> Optional: Specify an array of resources to include in the restore process. Resources might be shortcuts (for example, `po` for `pods`) or fully-qualified. If unspecified, all resources are included.
+<3> Optional: The `restorePVs` can be set to `false` in order to turn off restore of `PersistentVolumes` from `VolumeSnapshot` of Container Storage Interface (CSI) snapshots, or from native snapshots when `VolumeSnaphshotLocation` is configured.
```
[the remark this answers is not in the document]
Comment (CarmiWisemon) [reply modules/oadp-creating-restore-cr.adoc:43]: @bscott-rh 
In general in the documentation we don't use the expression "field".

[the remark this answers is not in the document]
Comment (bscott-rh) [reply modules/oadp-creating-restore-cr.adoc:43]: @CarmiWisemon how about "parameter"? we definitely need a noun after `restorePVs`.

[the remark this answers is not in the document]
Comment (bscott-rh) [reply modules/oadp-creating-restore-cr.adoc:43]:
```suggestion
<3> Optional: Disables restore of `PersistentVolumes` from `VolumeSnapshot` of Container Storage Interface (CSI) snapshots, or from native snapshots when `VolumeSnaphshotLocation` is configured.
```
What do you think about this phrasing instead?

[the remark this answers is not in the document]
Comment (CarmiWisemon) [reply modules/oadp-creating-restore-cr.adoc:43]: @bscott-rh 
I don't this is what DEV wanted written.
I'll add the phrase "parameter", as you requested earlier.

**Comment by CarmiWisemon** [comment]
@bscott-rh 
I added the phrase "parameter", please merge the PR.
Thank you for your help.
Carmi

**Pull request merged**